tests/Unit/Parallel/PhaseControl/Test_PhaseControlTags.cpp

```diff
@@ -0,0 +1,138 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "Framework/TestingFramework.hpp"
+
+#include <cstddef>
+#include <string>
+#include <type_traits>
+
+#include "DataStructures/DataBox/DataBoxTag.hpp"
+#include "Framework/TestCreation.hpp"
+#include "Helpers/DataStructures/DataBox/TestHelpers.hpp"
+#include "Options/Options.hpp"
+#include "Parallel/CharmPupable.hpp"
+#include "Parallel/PhaseControl/PhaseChange.hpp"
+#include "Parallel/PhaseControl/PhaseControlTags.hpp"
+#include "Parallel/RegisterDerivedClassesWithCharm.hpp"
+#include "Time/Triggers/Slabs.hpp"
+#include "Utilities/Functional.hpp"
+#include "Utilities/MakeString.hpp"
+#include "Utilities/TMPL.hpp"
+
+namespace {
+template <size_t val, typename PhaseChangeRegistrars>
+struct TestCreatable;
+
+namespace Registrars {
+template <size_t val>
+struct TestCreatable {
+  template <typename PhaseChangeRegistrars>
+  using f = ::TestCreatable<val, PhaseChangeRegistrars>;
+};
+}  // namespace Registrars
+
+namespace Tags {
+struct DummyDecisionTag1 {
+  using type = int;
+  using combine_method = funcl::Plus<>;
+  using main_combine_method = combine_method;
+};
+
+struct DummyDecisionTag2 {
+  using type = size_t;
+  using combine_method = funcl::Minus<>;
+  using main_combine_method = combine_method;
+};
+}  // namespace Tags
+
+template <size_t val, typename PhaseChangeRegistrars>
+struct TestCreatable : public PhaseChange<PhaseChangeRegistrars> {
+
+  TestCreatable() = default;
+  explicit TestCreatable(CkMigrateMessage* /*unused*/) noexcept {}
+  using PUP::able::register_constructor;
+  WRAPPED_PUPable_decl_template(TestCreatable);  // NOLINT
+
+  static std::string name() noexcept {
+    if constexpr(val == 1_st) {
+      return "TestCreatable(1)";
+    } else if constexpr(val == 2_st) {
+      return "TestCreatable(2)";
+    } else {
+      return "TestCreatable(Unknown)";
+    }
+  }
+
+  template <typename Metavariables>
+  using participating_components = tmpl::list<>;
+
+  static constexpr Options::String help{"Creatable for test"};
+
+  struct IntOption {
+    using type = int;
+    static constexpr Options::String help{"Option paramer"};
+  };
+
+  using options = tmpl::list<IntOption>;
+  using phase_change_tags_and_combines = tmpl::conditional_t<
+      1_st < val, tmpl::list<Tags::DummyDecisionTag1>,
```

Comment:
put `1_st < val` in `()` to improve readability. (`<` inside a template `<>` looks strange)